src/Migrations.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'node:path'
+import * as fsp from 'node:fs/promises'
+
+import { Database } from './Database'
+import { IMigrate } from './interfaces'
+
+import MigrationFile = IMigrate.MigrationFile
+import MigrationData = IMigrate.MigrationData
+
+const DEFAULT_MIGRATIONS_DIR_NAME = 'migrations'
+const DEFAULT_MIGRATIONS_DIR_PATH = path.join(
+  process.cwd(),
+  DEFAULT_MIGRATIONS_DIR_NAME
+)
+
+const MIGRATION_FILE_REGEXP = /^(\d+).(.*?)\.sql$/
+const MIGRATION_FILE_UP_REGEXP = /^-- .*$/gm
+const MIGRATION_FILE_DOWN_REGEXP = /^--\s+?down\b/im
+const MIGRATION_FILE_REPLACE_COMMAND_REGEXP = /^-{3,}$/gim
+
+export class Migrations<
+  Driver extends Database,
+  Config extends IMigrate.MigrationParams = IMigrate.MigrationParams
+> {
+  db: Driver
+  config: Config
```

Comment:
Did you mean for this to be IMigrate.MigrationParams?

Reply:
Yes, I updated it